frontend/src/stores/authSlice.ts

```diff
@@ -0,0 +1,48 @@
+import { setAsyncStorageItem } from '@/helpers/setAsyncStorageItem';
+import { authApi } from '@/services/authApi';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+type State = {
+  isLoggedIn: boolean;
+  userName: string;
+  id: string;
+};
+
+const initialState: State = {
+  isLoggedIn: false,
+  userName: '',
+  id: '',
+};
+
+const setUser = (state: State, action: PayloadAction<any>) => {
+  state.userName = action.payload.user.name;
+  state.id = action.payload.user.id;
+  state.isLoggedIn = true;
+  setAsyncStorageItem({
```

Comment:
is using asynchronous code inside a reducer a good practice? let's discuss this later together